tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/ADAuthTokenProvider.cs

```diff
@@ -0,0 +1,78 @@
+﻿
+using Microsoft.Identity.Client;
+using Newtonsoft.Json.Linq;
+using UKHO.ERPFacade.API.FunctionalTests.FunctionalTests;
+
+namespace UKHO.ERPFacade.API.FunctionalTests.Helpers
+{
+    public class ADAuthTokenProvider
+    {
+        static string AzureADToken = null;
+        public static Config _config;
+
+        public async Task<string> GetAzureADToken()
+        {
+            _config = new();
+
+            AzureADToken = await GenerateAzureADToken(_config.testConfig.AzureadConfiguration.ClientId, _config.testConfig.AzureadConfiguration.ClientSecret, AzureADToken);
+            return AzureADToken;
+        }
+
+        public async Task<string> GetAzureADToken(Boolean noRole)
+        {
+            _config = new();
+            if (noRole)
+            {
+                AzureADToken = null;
+                AzureADToken = await GenerateAzureADToken(_config.testConfig.AzureadConfiguration.AutoTestClientIdNoRole, _config.testConfig.AzureadConfiguration.ClientSecretNoRole, AzureADToken);
+            }
+            else
+            {
+                AzureADToken = await GenerateAzureADToken(_config.testConfig.AzureadConfiguration.AutoTestClientId, _config.testConfig.AzureadConfiguration.ClientSecret, AzureADToken);
+            }
+            return AzureADToken;
+        }
+
+        private static async Task<string> GenerateAzureADToken(string ClientId, string ClientSecret, string Token)
```

Comment:
Parameters should start lower-case, so 'clientId', 'clientSecret' etc

Reply:
incorporated changes.